FreeRTOS_IP.c

```diff
@@ -3011,6 +3017,173 @@ static eFrameProcessingResult_t prvProcessUDPPacket( NetworkBufferDescriptor_t *
+
+            /* Read the length expressed in number of octets. */
+            uxHopSize = ( size_t ) pucSource[ uxIndex + 1U ];
+            /* And multiply by 8 and add the minimum size of 8. */
+            uxHopSize = ( uxHopSize * 8U ) + 8U;
+
+            if( ( uxIndex + uxHopSize ) >= uxMaxLength )
+            {
+                uxIndex = uxMaxLength;
+                break;
+            }
+
+            uxIndex = uxIndex + uxHopSize;
+
+            if( ( ucNextHeader == ipPROTOCOL_TCP ) ||
+                ( ucNextHeader == ipPROTOCOL_UDP ) ||
+                ( ucNextHeader == ipPROTOCOL_ICMP_IPv6 ) )
+            {
+                FreeRTOS_printf( ( "Stop at header %u\n", ucNextHeader ) );
+                break;
+            }
+
+            xNextOrder = xGetExtensionOrder( ucNextHeader, pucSource[ uxIndex ] );
+
+/*
+ * [IP-Task         ] Going from header  0 (1) to 60 (2)
+ * [IP-Task         ] Going from header 60 (2) to 43 (3)
+ * [IP-Task         ] Going from header 43 (3) to 44 (4)
+ * [IP-Task         ] Stop at header 58
+ */
+            FreeRTOS_printf( ( "Going from header %2u (%d) to %2u (%d)\n",
+                               ucCurrentHeader,
+                               ( int ) xCurrentOrder,
+                               ucNextHeader,
+                               ( int ) xNextOrder ) );
+
+            if( xNextOrder <= xCurrentOrder )
+            {
+                FreeRTOS_printf( ( "Wrong order\n" ) );
+                uxIndex = uxMaxLength;
+                break;
+            }
+
+            ucCurrentHeader = ucNextHeader;
+            xCurrentOrder = xNextOrder;
+        }
+
+        if( uxIndex < uxMaxLength )
+        {
+            uint8_t * pucTo;
+            const uint8_t * pucFrom;
+            uint16_t usPayloadLength = FreeRTOS_ntohs( pxIPPacket_IPv6->xIPHeader.usPayloadLength );
+
+            uxRemovedBytes = uxIndex - ( ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER );
+
+            if( uxRemovedBytes >= ( size_t ) usPayloadLength )
+            {
+                /* Can not remove more bytes than the payload length. */
+            }
+            else if( xDoRemove == pdTRUE )
+            {
+                pxIPPacket_IPv6->xIPHeader.ucNextHeader = ucNextHeader;
+                pucTo = &( pxNetworkBuffer->pucEthernetBuffer[ ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER ] );
+                pucFrom = &( pxNetworkBuffer->pucEthernetBuffer[ uxIndex ] );
+                xMoveLen = uxMaxLength - uxIndex;
+                ( void ) memmove( pucTo, pucFrom, xMoveLen );
+                pxNetworkBuffer->xDataLength -= uxRemovedBytes;
+
+                usPayloadLength -= uxRemovedBytes;
+                pxIPPacket_IPv6->xIPHeader.usPayloadLength = FreeRTOS_htons( usPayloadLength );
+                eResult = eProcessBuffer;
+            }
+        }
+
+        FreeRTOS_printf( ( "Hop-by-hop option : %s Truncated %u bytes. Removed %u, Payload %u xDataLength now %u\n",
```

Comment:
```suggestion
        FreeRTOS_printf( ( "Extension headers: %s Truncated %u bytes. Removed %u, Payload %u xDataLength now %u\n",
```

Reply:
Yes you are right. In the first draft I mistakenly though that an extension header always starts with a hop-to-hop option.
I change the logging as suggested.